entries-api/support.js

```diff
@@ -89,3 +87,43 @@ const FILE_PATHS = [
   '/upload//file.txt',
   'subdir/./../file.txt',
 ];
+
+// ----------------------------------------
+// Helpers
+// ----------------------------------------
+
+// Wrapper for FileSystemDirectoryReader that yields all entries via a
+// Promise.
+
+function getEntriesAsPromise(dirEntry) {
+  return new Promise((resolve, reject) => {
+    let result = [];
+    let reader = dirEntry.createReader();
+    let doBatch = () => {
+      reader.readEntries(entries => {
+        if (entries.length > 0) {
+          entries.forEach(e => result.push(e));
```

Comment:
nit: isn't this just Array.prototype.push.apply(result, entries)? Or result = result.concat(entries); Although I guess neither of those is necessarily cleaner/simpler than what you have.

Reply:
All correct. Leaving as-is, though. I could make all those `let`s into `const`s though.